include/RAJA/policy/cuda/forall.hpp

```diff
@@ -193,6 +193,32 @@ RAJA_INLINE void forall_impl(cuda_exec<BlockSize, Async>,
 
 
 //
+////////////////////////////////////////////////////////////////////////
+//
+// Function templates for CUDA execution over iterables for within
+// and device kernel.  Called from RAJA::nested::*
+//
+////////////////////////////////////////////////////////////////////////
+//
+
+template <typename Iterable, typename LoopBody>
+RAJA_INLINE RAJA_DEVICE void forall_impl(cuda_loop_exec,
+                        Iterable&& iter,
+                        LoopBody&& loop_body)
+{
+  // TODO: we need a portable std::begin, std::end, and std::distance
```

Comment:
Is this comment still relevant?

Reply:
Not sure.  I had to resort to using seg.begin() and seg.end() instead of std::begin(seg), std::end(seg).

I think in this context, it may not matter.